### ISSUE TITLE
Add the same menu and static toolbar of lists for dataframe tables

### ISSUE THREAD

**Issue by dpturibio** [spyder-ide/ux-improvements#76]
Enhancement: Add the same menu and static toolbar of lists for dataframe tables.

![example](https://user-images.githubusercontent.com/19315417/194927042-343f24ad-d370-4ee0-9c85-706274c8fe76.png)

<b>Necessary changes</b><br>
Create a similar structure to BaseTableView of widgets/collectionseditor.py in DataFrameEditor of plugins/variableexplorer/widgets/dataframeeditor.py

![image](https://user-images.githubusercontent.com/19315417/194928217-c5b590dd-a61f-4c72-9468-e3cd99338c75.png)

Wold like to discuss about the possibility to apply this enhancement. Related issue: #67

**Comment by dpturibio** [comment]
Hello @dalthviz how are you?
There is some time since my last contribution, and as you guided me on that task (#67) , I am copying you to know your point of view regarding this new request.

**Comment by dalthviz** [comment]
Hi @dpturibio , hope everything has benn going well ! Regarding the idea seems quite interesting however seems like it will need way more work than #67 since, I believe, the logic for operations like insert values is not available for the DataFrameEditor widget. From a quick check, I think you can only change the dataframe values types or copy the values:

![dfe_ctx](https://user-images.githubusercontent.com/16781833/194934444-6de93552-a9e4-46f2-89ae-e0346e068dfa.png)

Just in case @spyder-ide/core-developers what do you think?

**Comment by ccordoba12** [comment]
Perhaps we could add the `Copy` and `Edit` actions. But what would be really useful is to add filters to each column.

**Comment by CAM-Gerlach** [comment]
Yeah, I recall at least one (by @jnsebgosselin ?) if not two PRs on that, but they never were completed and merged unfortunately. That would be super useful, but also would require some UI/UX work, which may not be so trivial...

**Comment by dpturibio** [comment]
Hello guys, how are you?
I've been working on this feature since last time, and I have some new things to share.
I created new features for DataFrame as edit, duplicate row/column, insert a row/column after/before/above/bellow the selected one, remove, and resize as you can see in screenshot.

![image](https://user-images.githubusercontent.com/19315417/218728065-36aa55c5-c4d9-4255-83c5-0c3922d9590e.png)



I was checking documentation and found that branch 5.x is only for bugfixes and new features I should use master.
That's correct? So I should use master for this kind of changes proposed?

Let me know so I can upload the code.

**Comment by ccordoba12** [comment]
Hey @dpturibio, that looks great! Thanks for all the work you put into that!

> I was checking documentation and found that branch 5.x is only for bugfixes and new features I should use master.
That's correct? So I should use master for this kind of changes proposed?

Yep, that's correct. Spyder 5 has entered the features freeze period and now we're trying to focus only on important bugfixes for it. So, any enhancement or additional functionality needs to go to master, which will be part of Spyder 6, to be released during the second half of the year.

If you already created your branch against `5.x`, you need to run the following commands to move it to master:

    git checkout <pr_branch>
    git rebase --onto master 5.x <pr_branch>
    git push -f origin <pr_branch>

**Comment by CAM-Gerlach** [comment]
(You might want to make a backup copy of the branch first, with e.g. `git branch -c <pr-branch> <pr-branch>-copy`, so you have it easy to access just in case.)

**Comment by dpturibio** [comment]
Thank you @ccordoba12 and @CAM-Gerlach , I created a [PR](https://github.com/spyder-ide/spyder/pull/20546) with the changes.
Next step I think is the review by someone of the team right?
If you can quickly explain me what is the procedure when a PR is submitted, who assigns to it, is it all automatic from the point of view of the developer(me), or things happen accordingly to a conversation in issue page like this? I am just wondering how you manage all the work, curious to understand :grinning:.

**Comment by CAM-Gerlach** [comment]
Besides letting us know here, which is helpful, we'll take it from there, review it and let you know if it needs any refinements. In case you don't hear from anyone after a week or two, feel free to give us another ping.
